tests/test_deadfixtures.py

```diff
@@ -237,3 +237,29 @@ def test_conftest_fixture():
 
     assert result.ret == 0
     assert message not in result.stdout.str()
+
+
+def test_do_not_list_fixture_used_after_test_which_does_not_use_fixtures(testdir, message_template):
+    testdir.makepyfile("""
+        import pytest
+
+
+        @pytest.fixture()
+        def same_file_fixture():
+            return 1
+
+        def test_no_fixture_used():
+            assert True
+
+        def test_simple(same_file_fixture):
+            assert 1 == same_file_fixture
+    """)
+
+    result = testdir.runpytest('--dead-fixtures')
+    message = message_template.format(
+        'same_file_fixture',
+        'test_dont_list_fixture_used_after_test_which_does_not_use_fixtures'
```

Comment:
the test name here should be the same as the "outer test", in this case `test_do_not_list_fixture_used_after_test_which_does_not_use_fixtures`